sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp

```diff
@@ -497,6 +497,52 @@ namespace Azure { namespace Storage { namespace Blobs {
     Azure::Nullable<bool> ShouldSealDestination;
   };
 
+  /**
+   * @brief Optional parameters for #Azure::Storage::Blobs::BlobClient::CopyFromUri.
+   */
+  struct CopyBlobFromUriOptions
+  {
+    /**
+     * @brief Specifies user-defined name-value pairs associated with the blob. If no
+     * name-value pairs are specified, the operation will copy the metadata from the source blob or
+     * file to the destination blob. If one or more name-value pairs are specified, the destination
+     * blob is created with the specified metadata, and metadata is not copied from the source blob
+     * or file.
+     */
+    Storage::Metadata Metadata;
+
+    /**
+     * @brief The tags to set for this blob.
+     */
+    std::map<std::string, std::string> Tags;
+
+    /**
+     * @brief Optional conditions that must be met to perform this operation.
+     */
+    BlobAccessConditions AccessConditions;
+
+    /**
+     * @brief Optional conditions that the source must meet to perform this operation.
+     *
+     * @note Lease access condition only works for API versions before 2012-02-12.
+     */
+    struct : public Azure::ModifiedConditions, public Azure::MatchConditions
+    {
+    } SourceAccessConditions;
```

Comment:
This is an interesting code style. Do we do this elsewhere? And why do we have this empty struct?

1) Unnamed struct, with name after?
2) Struct that inherits from multiple others (rather than class)?

Reply:
> Do we do this elsewhere?

Yes, mainly for various combinations of access conditions.

>  And why do we have this empty struct?

It's not empty. It inherits from two other structs, which makes it effectively a union of the two structs.

> Unnamed struct, with name after?

It has the same effect as

```C++
struct StructWillOnlyBeUsedOnce { // bla bla };
StructWillOnlyBeUsedOnce SourceAccessConditions;
```

except that `StructWillOnlyBeUsedOnce` is not accessible elsewhere.

> Struct that inherits from multiple others (rather than class)?

All its members are public and it works like a struct. So I think `struct` is better than `class` here.

Reply:
This is a convenient way to define structs that will only be used once, and it can help us avoid the trouble of naming things.

For example, how would you want to name this struct with four access conditions? Another operation supports only 3 access conditions (`IfMatch`, `IfNoneMatch`, `IfModifiedSince`), how do you want to name that? With anonymous struct, we don't need to worry about that at all.

Reply:
Hmm, interesting. I was thinking about the rules/logic on when and how such a struct definition would be code-gen'd.

Reply:
Generate like this

```C++
struct CopyBlobFromUriOptions {
  Storage::Metadata Metadata;
  std::map<std::string, std::string> Tags;
  BlobAccessConditions AccessConditions;
  (anonymous123) SourceAccessConditions;
  ...
};
```

and `(anonymous123)` has a link to the definition of the type, so if you click on it, it jumps to

```C++
struct (anonymous123) : public Azure::ModifiedConditions, public Azure::MatchConditions
{
};
```